### ISSUE TITLE
service port is named "https" also when TLS is disabled

### ISSUE THREAD

**Issue by trondhindenes** [elastic/cloud-on-k8s#2216]
## Proposal
If disabling SSL using the spec:
```
spec:
  http:
    tls:
      selfSignedCertificate:
        disabled: true
```
The resulting `service` object's port name will still be `https`. This is a problem, as it means that a regular http (non-ssl) port is named https, indicating that it's a https port.

Traefik uses the port name to determine if it should proxy requests using http or https. The current ECK behavior breaks Traefik, and requires us to manually create a service with correct naming, something we'd like to avoid.

Suggestion: when ssl is disabled, make the port name `http` instead of `https`.

**Comment by charith-elastic** [comment]
This issue is fixed in master by #2083.

**Comment by trondhindenes** [comment]
Sorry, I should have checked for duplicate issues. Thanks!
